### PR TITLE
Added 8-bit weight compression for OVModel

### PR DESCRIPTION
**Pull request by l-bat**
# What does this PR do?

Introduced data-free weight compression to 8 bits for `OVBaseModel` and `OVBaseDecoderModel`
PR to nncf https://github.com/openvinotoolkit/nncf/pull/2059 was merged

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Review by echarlaix — APPROVED** [review]
LGTM, thanks for the addition @l-bat

**Comment by l-bat** [comment]
@echarlaix there are some failed tests not related to my changes

File: tests/openvino/test_quantization.py [hunk 1]
```diff
@@ -146,12 +146,12 @@ def preprocess_function(examples, tokenizer):
 class OVWeightCompressionTest(unittest.TestCase):
     # TODO : add models
     SUPPORTED_ARCHITECTURES_WITH_EXPECTED_COMPRESSED_MATMULS = (
-        (OVModelForSequenceClassification, "hf-internal-testing/tiny-random-bert", 70),
-        (OVModelForCausalLM, "hf-internal-testing/tiny-random-gpt2", 45),
+        (OVModelForSequenceClassification, "hf-internal-testing/tiny-random-bert", 70, 35),
+        (OVModelForCausalLM, "hf-internal-testing/tiny-random-gpt2", 45, 22),
```
Comment (echarlaix) [inline tests/openvino/test_quantization.py:150]: What is the difference in the quantization applied on the two models (depending on whether this is a pytorch or an openvino model) ?

**Review by echarlaix — APPROVED** [review]
Thanks for iterating on it, the PR looks good so will merge, also added a question on the changes applied in https://github.com/huggingface/optimum-intel/pull/415/commits/6cd4a85a0981886c48d5fe0c5c412c7f8aeb247c

**Pull request merged**